layer/executables/packager.py

```diff
@@ -0,0 +1,154 @@
+import inspect
+import os
+import pickle  # nosec
+import shutil  # nosec
+import sys
+import tempfile
+import zipapp
+from pathlib import Path
+from typing import Any, Callable, List, Optional
+from uuid import uuid4
+
+import cloudpickle  # type: ignore
+
+
+def package_function(
+    function: Callable[..., Any],
+    resources: Optional[List[Path]] = None,
+    pip_dependencies: Optional[List[str]] = None,
```

Comment:
Perhaps it's a concern of the users of this packager.py file, but https://docs.app.layer.ai/docs/sdk-library/pip-requirements-decorator also supports a file too (not just list of requirement strings)